posts/running-coder-in-k3s/index.md

```diff
@@ -0,0 +1,507 @@
+
+```yaml
+coder:
+  env:
+    - name: CODER_PG_CONNECTION_URL
+      valueFrom:
+        secretKeyRef:
+          name: coder-db-url
+          key: url
+
+    - name: CODER_ACCESS_URL
+      # TODO
+      value: "https://coder.mydomain.com"
+    - name: CODER_WILDCARD_ACCESS_URL
+      # TODO
+      value: "*.coder.mydomain.com"
+
+  service:
+    enable: true
+    type: ClusterIP
+    sessionAffinity: ClientIP
+    externalTrafficPolicy: Cluster
+    loadBalancerIP: ""
+    annotations: {}
+
+  ingress:
+    enable: true
+    className: ""
+    # TODO
+    host: "coder.mydomain.com"
+    # TODO
+    wildcardHost: "*.coder.mydomain.com"
+    annotations: {}
+    tls:
+      enable: false
+      secretNames: ""
+      wildcardSecretName: ""
+```
+
+The service will be configured as a ClusterIP. We configure Ingress to handle requests for the (`coder.mydomain.com`) domain.
+
+* Install the HELM chart on your [K3s](https://k3s.io/) cluster:
+
+```console
+$ helm install coder coder-v2/coder --namespace coder --values values.yaml
+...
+```
+
+* Check the pods into the `coder` namespace:
+
+```console
+$ kubectl get pods --namespace coder
+NAME                     READY   STATUS    RESTARTS   AGE
+coder-db-postgresql-0    1/1     Running   0          22d
+coder-59c6bc9c77-6f2wj   1/1     Running   0          9m47s
+```
+
+> These instructions are based on the [Coder](https://coder.com/) website: https://coder.com/docs/v2/latest/install/kubernetes.
+
+## Deploy Reverse Proxy
+
+As mentioned in the introduction, a Reverse Proxy will be deployed outside of your Kubernetes cluster.
+
+The Reverse Proxy will also be in charge of managing SSL/TLS certificates. Let's describe how to generate certificates with LetsEncrypt.
+
+* Connect to the server node (`k3sserver`) and install Certbot:
+
+```console
+$ sudo apt-get update 
+$ sudo apt-get install certbot -y
+```
+
+* Create the SSL/TLS certificates:
+
+```console
+$ sudo certbot certonly --agree-tos -m YOUR_EMAIL --manual --preferred-challenges=dns -d 'coder.mydomain.com' -d '*.coder.mydomain.com' -v
+...
+```
+
+* Copy the SSL/TLS certificates files (_fullchain.pem_ and _privkey.pem_) into a directory (i.e. _/ssl_):
+
+```console
+$ mkdir /ssl
+$ cp /etc/letsencrypt/live/coder.mydomain.com-0001/fullchain.pem privkey.pem /ssl
+```
+
+* Generate _dhparams.pem_ file:
+
+```console
+$ cd /ssl
+$ openssl dhparam -out dhparams.pem 4096
+```
+
+You need to configure your Kubernetes cluster to update HTTP and HTTPS listen ports.
+
+* Connect to the server node (`k3sserver`) and create a _/var/lib/rancher/k3s/server/manifests/traefik-config.yaml_ file with the following content:
+
+```yaml
+kind: HelmChartConfig
+metadata:
+  name: traefik
+  namespace: kube-system
+spec:
+  valuesContent: |-
+    ports:
+      web:
+        exposedPort: 8080
+      websecure:
+        exposedPort: 8443
+```
+
+* Apply this configuration:
```

Comment:
While personally going through this guide, I got stuck at this point.
As a suggestion, why do you prefer a reverse-proxy outside K3s instead of using a cert-manager?

Reply:
My institution supplies a reverse proxy (it is a restriction). I would like reproduce the same setup. An external reverse proxy which listens HTTPS query and sends to the Kubernetes cluster HTTP queries.